greenswitch/esl.py

```diff
@@ -252,6 +252,13 @@ def authenticate(self):
         if response.headers['Reply-Text'] != '+OK accepted':
             raise ValueError('Invalid password.')
 
+    def __enter__(self):
+        self.connect()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.send('exit')
```

Comment:
you can call only the stop method here, the stop method does join of some greenlets and closes the socket connections too.